x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1006,39 +1009,46 @@ public LogicalPlan apply(LogicalPlan p) {
         }
     }
 
+    private static class PercentileAggKey extends Tuple<Expression, PercentilesConfig> {
+        PercentileAggKey(HasPercentilesConfig per) {
+            super(per.field(), per.percentilesConfig());
+        }
+        
+        public Expression field() {
+            return v1();
+        }
+        
+        public PercentilesConfig percentilesConfig() {
+            return v2();
+        }
+    }
```

Comment:
Percentile is already an agg making `PercentileAgg` redundant - just call the tuple `PercentileKey`.
Also the methods should be all private/protected where possible.